src/pages/task/ActionButtons.tsx

```diff
@@ -0,0 +1,248 @@
+import React, { useRef, useState, useEffect } from "react";
+import { Button } from "components/Button";
+import styled from "@emotion/styled";
+import { useOnClickOutside } from "hooks";
+import { InputNumber, Popconfirm } from "antd";
+import get from "lodash/get";
+import { Body } from "@leafygreen-ui/typography";
+import { useParams } from "react-router-dom";
+import { useMutation } from "@apollo/react-hooks";
+import { ABORT_TASK } from "gql/mutations/abort-task";
+import { RESTART_TASK } from "gql/mutations/restart-task";
+import { SCHEDULE_TASK } from "gql/mutations/schedule-task";
+import { UNSCHEDULE_TASK } from "gql/mutations/unschedule-task";
+import { SET_TASK_PRIORTY } from "gql/mutations/set-task-priority";
+import { useBannerDispatchContext } from "context/banners";
+import { ButtonRow } from "components/ButtonRow";
+import {
+  SetTaskPriorityMutation,
+  SetTaskPriorityMutationVariables,
+  AbortTaskMutation,
+  AbortTaskMutationVariables,
+  RestartTaskMutation,
+  RestartTaskMutationVariables,
+  ScheduleTaskMutation,
+  ScheduleTaskMutationVariables,
+  UnscheduleTaskMutation,
+  UnscheduleTaskMutationVariables,
+} from "gql/generated/types";
+import { CardItem } from "components/CardDropdownButton";
+
+interface Props {
+  priority?: number;
+  canAbort: boolean;
+  canRestart: boolean;
+  canSchedule: boolean;
+  canUnschedule: boolean;
+  canSetPriority: boolean;
+}
+
+export const ActionButtons = ({
+  canAbort,
+  canRestart,
+  canSchedule,
+  canSetPriority,
+  canUnschedule,
+  ...props
+}: Props) => {
+  const { success, error } = useBannerDispatchContext();
+  const wrapperRef = useRef(null);
+  const priorityRef = useRef(null);
+  const [isVisible, setIsVisible] = useState(false);
+  const [priority, setPriority] = useState<number>(props.priority);
+  const { id: taskId } = useParams<{ id: string }>();
+
+  const [scheduleTask, { loading: loadingScheduleTask }] = useMutation<
+    ScheduleTaskMutation,
+    ScheduleTaskMutationVariables
+  >(SCHEDULE_TASK, {
+    variables: { taskId },
+    onCompleted: () => {
+      success("Task marked as scheduled");
```

Comment:
Should we clear the error on success.

Reply:
This seems like a good idea because the error is no longer relevant if the action succeeds but we should make sure with design first.

Reply:
this will require changes in the banners component as well as a few other pages + tests so I created a separate ticket for this: https://jira.mongodb.org/browse/EVG-7973

Reply:
we don't need to clear things like this with design first i don't think. the error is meant to go away after it has succesfully been resolved. 

clearing the errors will require clearing all the banners on the page. currently there is no way to remove a specific banner from the component level. you would need to store the error's id in the component state to do this.